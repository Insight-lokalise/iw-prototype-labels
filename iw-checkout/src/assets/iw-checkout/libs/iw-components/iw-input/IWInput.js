import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

import { FieldLabel, FieldErrorMessage } from './'

export default function IWInput(props) {
  const isCheckbox = props.type === 'checkbox'
  const inputClassName = cn('form__field', props.className, {
    'form__field--error': props.showError,
    'form__input--checkbox': isCheckbox,
  })

  const {
    children,
    checked,
    className,
    customInputProps,
    disabled,
    errorMessage,
    hideLabel,
    label,
    maxLength,
    name,
    placeholder,
    readOnly,
    regexFormat,
    required,
    showError,
    showHelpIcon,
    tooltip,
    type,
    ...inputAttributes
  } = props

  return (
    <div>
      <FieldLabel
        className={className}
        hideLabel={hideLabel}
        label={label}
        name={name}
        regexFormat={regexFormat}
        required={required}
        showHelpIcon={showHelpIcon}
        tooltip={tooltip}
      >
        {type === 'textArea' ? (
          <textarea
            className={inputClassName}
            disabled={disabled}
            maxLength={maxLength}
            placeholder={placeholder}
            readOnly={readOnly}
            {...customInputProps}
            {...inputAttributes}
          />
        ) : (
          <input
            className={inputClassName}
            checked={checked}
            disabled={disabled}
            formNoValidate
            maxLength={maxLength}
            placeholder={placeholder}
            readOnly={readOnly}
            type={type}
            {...customInputProps}
            {...inputAttributes}
          />
        )}
      </FieldLabel>
      {children}
      <FieldErrorMessage errorMessage={errorMessage} showError={showError} />
    </div>
  )
}

IWInput.propTypes = {
  // required
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['checkbox', 'email', 'file', 'tel', 'text', 'textArea']).isRequired,
  // optional
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  className: PropTypes.string,
  customInputProps: PropTypes.shape({}),
  disabled: PropTypes.bool,
  errorMessage: PropTypes.string,
  hideLabel: PropTypes.bool,
  maxLength: PropTypes.number,
  placeholder: PropTypes.string,
  readOnly: PropTypes.bool,
  regexFormat: PropTypes.string,
  required: PropTypes.bool,
  showError: PropTypes.bool,
  showHelpIcon: PropTypes.bool,
  tooltip: PropTypes.node,
  // checkbox and radio specific
  checked: PropTypes.bool,
}

IWInput.defaultProps = {
  children: null,
  checked: false,
  className: '',
  customInputProps: {},
  disabled: false,
  errorMessage: '',
  hideLabel: false,
  placeholder: '',
  readOnly: false,
  regexFormat: '',
  required: false,
  showError: false,
  showHelpIcon: false,
  tooltip: null,
}
