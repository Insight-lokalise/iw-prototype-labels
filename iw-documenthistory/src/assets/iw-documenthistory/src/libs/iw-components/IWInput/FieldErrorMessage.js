import React from 'react'
import PropTypes from 'prop-types'

export default function FieldErrorMessage(props) {
  const emptyString = ''
  return (
    <div className="form__field-msg form__field-msg--error">{props.showError ? props.errorMessage : emptyString}</div>
  )
}

FieldErrorMessage.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  showError: PropTypes.bool.isRequired,
}
