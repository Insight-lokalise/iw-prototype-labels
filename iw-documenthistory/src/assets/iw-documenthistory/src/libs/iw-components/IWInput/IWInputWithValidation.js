import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { IWInput } from './'

export default class IWInputWithValidation extends Component {
  constructor() {
    super()
    this.state = { showError: false }

    this.handleTyping = this.handleTyping.bind(this)
    this.validate = this.validate.bind(this)
  }

  handleTyping(e) {
    const { value, defaultValue } = e.target
    value !== defaultValue && this.validate(value)
  }

  hideError() {
    this.setState({ showError: false })
  }

  showError() {
    this.setState({ showError: true })
  }

  validate(value) {
    !this.props.isValid(value) ? this.showError() : this.hideError()
  }

  render() {
    const { defaultValue, errorMessage, name, label, type, ...otherProps } = this.props
    return (
      <IWInput
        defaultValue={defaultValue}
        errorMessage={errorMessage}
        label={label}
        name={name}
        onChange={this.handleTyping}
        onBlur={e => this.validate(e.target.value)}
        showError={this.state.showError}
        type={type}
        {...otherProps}
      />
    )
  }
}

IWInputWithValidation.propTypes = {
  defaultValue: PropTypes.string,
  errorMessage: PropTypes.string.isRequired,
  isValid: PropTypes.func,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['checkbox', 'email', 'file', 'tel', 'text', 'textArea']).isRequired,
}

IWInputWithValidation.defaultProps = {
  defaultValue: '',
  isValid: () => true,
}
