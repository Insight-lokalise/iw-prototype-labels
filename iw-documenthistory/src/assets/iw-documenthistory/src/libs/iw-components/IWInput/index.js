export { default as IWInput } from './IWInput'
export { default as FieldLabel } from './FieldLabel'
export { default as FieldErrorMessage } from './FieldErrorMessage'
export { default as IWInputWithValidation } from './IWInputWithValidation'
export { MaxLength } from './inputSFCs'
