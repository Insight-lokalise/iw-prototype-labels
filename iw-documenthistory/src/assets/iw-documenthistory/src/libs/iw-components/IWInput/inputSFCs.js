import React from 'react'
import PropTypes from 'prop-types'
import { t } from '@insight/toolkit-utils/lib/labels'

export function IWAsterisk() {
  const asteriskText = '*'
  return <span className="form__required">{asteriskText} </span>
}

export function MaxLength(props) {
  return (
    <div className="form__help-text text-right">
      {`${props.maxLength - (props.currentLength || 0)} ${t('characters remaining')}`}
    </div>
  )
}

MaxLength.propTypes = {
  currentLength: PropTypes.number.isRequired,
  maxLength: PropTypes.number.isRequired,
}
