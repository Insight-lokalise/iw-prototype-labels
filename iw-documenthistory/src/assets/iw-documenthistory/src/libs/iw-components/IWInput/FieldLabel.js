import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'
import { t } from '@insight/toolkit-utils/lib/labels'
import IWHelpIcon from './../IWTooltip/IWHelpIcon'

import { IWAsterisk } from './inputSFCs'

export default function FieldLabel(props) {
  const labelText = props.hideLabel ? null : `${props.label} `
  const isCheckbox = props.type === 'checkbox'
  return (
    <label className={cn('form__label', props.className)} htmlFor={props.name}>
      {isCheckbox && props.children}
      <span
        className={cn(
          { 'form__label--checkbox': isCheckbox, hide: props.hideLabel },
          'form__label-text',
          props.labelClassName
        )}
      >
        {labelText}
        {props.required && <IWAsterisk />}
        {!!props.regexFormat && ` (${t('format')}: ${props.regexFormat}) `}
        {props.showHelpIcon && <IWHelpIcon className={props.tooltipClass} tooltip={props.tooltip} />}
      </span>
      {!isCheckbox && props.children}
    </label>
  )
}

FieldLabel.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  className: PropTypes.string,
  labelClassName: PropTypes.string,
  hideLabel: PropTypes.bool,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  regexFormat: PropTypes.string,
  required: PropTypes.bool.isRequired,
  showHelpIcon: PropTypes.bool,
  tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  tooltipClass: PropTypes.string,
  type: PropTypes.string,
}

FieldLabel.defaultProps = {
  children: null,
  className: '',
  labelClassName: '',
  hideLabel: false,
  regexFormat: '',
  showHelpIcon: false,
  tooltip: '',
  tooltipClass: '',
  type: 'text',
}
