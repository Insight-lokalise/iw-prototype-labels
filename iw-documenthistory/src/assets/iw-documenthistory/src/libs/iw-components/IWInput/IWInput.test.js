import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

import IWInput from './IWInput'

describe('IWInput', () => {
  it('renders checkbox correctly', () => {
    const {asFragment} = render(<IWInput label="Testing" name="testing" type="checkbox" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders textArea correctly', () => {
    const {asFragment} = render(<IWInput label="Testing" name="testing" type="textArea" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders tel correctly', () => {
    const {asFragment} = render(<IWInput label="Testing" name="testing" type="tel" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
