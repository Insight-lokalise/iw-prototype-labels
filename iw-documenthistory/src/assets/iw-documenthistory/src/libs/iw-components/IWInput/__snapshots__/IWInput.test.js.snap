// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IWInput renders checkbox correctly 1`] = `
<DocumentFragment>
  <div>
    <label
      class="form__label"
      for="testing"
    >
      <span
        class="form__label-text"
      >
        Testing 
      </span>
      <input
        class="form__field form__input--checkbox"
        formnovalidate=""
        placeholder=""
        type="checkbox"
      />
    </label>
    <div
      class="form__field-msg form__field-msg--error"
    />
  </div>
</DocumentFragment>
`;

exports[`IWInput renders tel correctly 1`] = `
<DocumentFragment>
  <div>
    <label
      class="form__label"
      for="testing"
    >
      <span
        class="form__label-text"
      >
        Testing 
      </span>
      <input
        class="form__field"
        formnovalidate=""
        placeholder=""
        type="tel"
      />
    </label>
    <div
      class="form__field-msg form__field-msg--error"
    />
  </div>
</DocumentFragment>
`;

exports[`IWInput renders textArea correctly 1`] = `
<DocumentFragment>
  <div>
    <label
      class="form__label"
      for="testing"
    >
      <span
        class="form__label-text"
      >
        Testing 
      </span>
      <textarea
        class="form__field"
        placeholder=""
      />
    </label>
    <div
      class="form__field-msg form__field-msg--error"
    />
  </div>
</DocumentFragment>
`;
